bin/zerobufCxx.py

```diff
@@ -962,10 +966,8 @@ def parse_members(self, fbsFile):
                 else:
                     elem_count = int(attrib[4])
                     member = FixedSizeArray(name, value_type, elem_count, self.name)
-                    if is_byte_type:
-                        property['type'] = 'string'
-                        property['media'] = {}
-                        property['media']['binaryEncoding'] = 'base64'
+                    if is_byte_type: # byte arrays are base64 strings
+                        property = _add_base64_string(property)
```

Comment:
I just discovered that it is more subtle than that, you can't really reason in C++ terms of copy or reference with the python 'variables', they are more like 'labels' pointing to objects which are either mutable, or immutable. In this case 'property' is mutable (it's like a list) and it is modified by the function - the assignment and return value have no effect (it just reassigns 'property' to 'property'). Some references:
http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html#other-languages-have-variables
http://stackoverflow.com/questions/986006/how-do-i-pass-a-variable-by-reference/8140747#8140747